ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,235 @@
+import os, shutil, time, json
+import contextlib2
+import tensorflow as tf
+from pathlib import Path
+from random import shuffle
+from datetime import datetime
+from ravenml.data.interfaces import CreateInput
+from ravenml.utils.question import cli_spinner, cli_spinner_wrapper
+from ravenml.utils.config import get_config
+from ravenml.data.helpers import default_filter, default_load_image_ids, copy_data_locally
+
+class DatasetWriter(object):
+
+    def __init__(self, create: CreateInput, **kwargs):
+        self.associated_files = kwargs['associated_files']
+        self.related_data_prefixes = kwargs['related_data_prefixes']
+        
+        metadata = create.metadata
+        self.num_folds = create.kfolds # Never actually used
+        self.test_percent = create.test_percent
+        self.label_to_int_dict = {}
```

Comment:
This isn't being used at all

Reply:
It is being used by the bbox plugin (in the construct and export methods). Do you think it is something that could be used generically by the DefaultDatasetWriter class, or should I move it entirely to the bbox plugin?